LEGO1/lego/legoomni/src/worlds/hospital.cpp

```diff
@@ -228,11 +230,149 @@ MxLong Hospital::HandleKeyPress(MxS8 p_key)
 	return result;
 }
 
-// STUB: LEGO1 0x10074e00
+// FUNCTION: LEGO1 0x10074e00
 MxLong Hospital::HandleEndAction(MxEndActionNotificationParam& p_param)
 {
-	// TODO
-	return 0;
+	MxLong result = 0;
+	MxDSAction* action = p_param.GetAction();
+	Act1State* act1State;
+
+	if (action->GetAtomId() != m_atom) {
+		return result;
+	}
+
+	m_unk0x108 = 0;
+
+	switch (m_hospitalState->m_unk0x08.m_unk0x00) {
+	case 5:
+		m_hospitalState->m_unk0x08.m_unk0x00 = 7;
+		PlayAction(HospitalScript::c_hho006cl_RunAnim);
+
+		m_unk0x10c = HospitalScript::c_hho006cl_RunAnim;
+		m_unk0x108 = 1;
+		m_unk0x118 = 1;
+		g_unk0x100f7918 = 0;
+		break;
+	case 6:
+		m_time = Timer()->GetTime();
+		m_unk0x100 = 1;
+		break;
+	case 7:
+	case 10:
+		m_hospitalState->m_unk0x08.m_unk0x00 = 8;
+		m_unk0x100 = 1;
+		m_time = Timer()->GetTime();
+		break;
+	case 11:
+		switch (m_currentActorId) {
+		case 1:
+			if (0 > m_hospitalState->m_unk0x0e || 1 >= m_hospitalState->m_unk0x0e) {
```

Comment:
The conditions in all of those branches are incorrect. If you look closely at the assembly you will notice the jump offsets don't match the original